src/trace.ml

```diff
@@ -561,6 +563,41 @@ module Make_commands (Backend : Backend_intf.S) = struct
     { Decode_opts.output_config; decode_opts; print_events }
   ;;
 
+  module Enable = struct
+    type enabled = { value : string }
+
+    type t =
+      | Disabled
+      | Enabled of enabled
+
+    let processor_param =
+      match Env_vars.processor_path with
+      | None -> Command.Param.return Disabled
+      | Some processor_shell_path ->
+        let%map_open.Command processor =
+          flag
+            "use-trace-processor-shell"
+            no_arg
+            ~doc:[%string "use the trace processor set in environment variables"]
+        in
+        if processor then Enabled { value = processor_shell_path } else Disabled
+    ;;
+  end
+
+  (* Same as [Caml.exit] but does not run at_exit handlers *)
+  external sys_exit : int -> 'a = "caml_sys_exit"
+
+  let call_trace_processor ?env ~prog ~argv () =
```

Comment:
Let's either generalize this with the `perf` version, or leave a `CR-someday` to do this later.

Reply:
I am unsure which part of this function is perf-specific.

Reply:
None of it is, so ideally this'd be the same function as https://github.com/janestreet/magic-trace/blob/d51d797dfd197367423a402b8fae1d56f46bc329/src/perf_tool_backend.ml#L70 so we don't have two copies.

You don't need to do this in this PR, but it would be good to leave a note for the future.